### ISSUE TITLE
feature: different speed limit for LAN->LAN connections

### ISSUE THREAD

**Issue by ghost** [transmission/transmission#1310]
Hello,...

I often need to set speed limit, so my Internet is not overloaded. But because of that, I'm downloading slower even in situation, when transfer is not going over Internet but inside local network (LAN).

Is there way how to ignore speed limits only for LAN->LAN connections? (we can just detect if IP of my partner is same as my, or add flag if connection was created over LPD)

**Comment by tuxuday5** [comment]
Can't per torrent limits be used in your case?

**Comment by Mlocik97-issues** [comment]
No, because one torrent can have seeders from local network as well as from Internet. So it's solving nothing.

**Comment by ckerr** [comment]
My main problem with this is that it seems like it would be almost trivial to fix outside of Transmission. If you're wanting unlimited speeds on a LAN, why not use rsync to sync between machines?

Even if it _does_ need to be done at the BitTorrent level -- e.g. if you've got a farm of clients all downloading the same torrent because you have a bot to jump on new downloads -- that seems like a use case maybe 50 people will ever have. If the project said yes to every request like this, the Preferences / Settings dialog would be 100 pages long and all the maintainers would be dead. :skull: 

**If** someone wants to do the legwork on this and **if** the PR is simple and easy to maintain and **if** it is only exposed in the settings.json config file, i.e. no maintenance cost in the GUI apps -- I'd be willing to review the PR and possibly merge it.

**Comment by ckerr** [comment]
Closing as 'declined' since nobody else has requested this feature in 5 years + it would require some complex logic.

The idea has some merit & I'd be willing to review if someone wants to write this & put up a PR, but the code for this is  probably not going to come from the maintainers.
